crates/hir-ty/src/infer/closure/analysis.rs

```diff
@@ -0,0 +1,1298 @@
+//! Post-inference closure analysis: captures and closure kind.
```

Comment:
This file was just moved from `crates/hir-ty/src/infer/closure.rs`, to clean things up and also to not mix new solver and Chalk imports.